lightning-transaction-sync/Cargo.toml

```diff
@@ -0,0 +1,32 @@
+[package]
+name = "lightning-transaction-sync"
+version = "0.0.113"
+authors = ["Elias Rohrer"]
+license = "MIT OR Apache-2.0"
+repository = "http://github.com/lightningdevkit/rust-lightning"
+description = """
+Utilities for syncing LDK via the transaction-based `Confirm` interface.
+"""
+edition = "2018"
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
+
+[features]
+default = ["esplora-blocking"]
```

Comment:
I think `esplora-blocking` should not be a default feature. For people using the async API it's easy to forget to specify `default-features = false` when declaring `lightning-transaction-sync`, which results in accidentally including the whole blocking dependency tree which they do not intend to use.

I made this mistake myself while using the `lightning-transaction-sync` crate, trying to figure out why `ureq` was being compiled into our project. `ldk-node` https://github.com/lightningdevkit/ldk-node/pull/11 makes this mistake as well; including the blocking dependency tree in the library even though it uses only the async API.

Reply:
Mh, good point. While having a sane default would be nice (and we may even get away with defaulting to `blocking` for the time being), it will def. stop making sense when the user has to decide between Esplora or Electrum variant of the crate anyways. I therefore now removed the default.

Apart from that happy to hear you're looking into using this crate. Please let me know if you find anything else noteworthy!